api/topicservice.yml

```diff
@@ -0,0 +1,143 @@
+          description: BAD REQUEST - invalid/incomplete request or body object
+        401:
+          description: UNAUTHORIZED - no/invalid Keycloak token
+        500:
+          description: INTERNAL SERVER ERROR - server encountered unexpected condition
+    get:
+      tags:
+        - topic-controller
+      summary: 'Get all topics'
+      operationId: getAllTopics
+      responses:
+        200:
+          description: OK - successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/TopicDTO'
+        204:
+          description: NO CONTENT - no content found
+        400:
+          description: BAD REQUEST - invalid/incomplete request or body object
+        401:
+          description: UNAUTHORIZED - no/invalid Keycloak token
+        500:
+          description: INTERNAL SERVER ERROR - server encountered unexpected condition
+  /topic/{id}:
+    summary: Represents a topic
+    description: This resource represents an individual topic in a system.
+    get:
+      tags:
+        - topic-controller
+      summary: 'Gets a topic by its id [Authorization: Role: topic-admin]'
+      operationId: getTopicById
+      parameters:
+        - name: id
+          in: path
+          description: Topic ID
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        200:
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TopicDTO'
+        400:
+          description: BAD REQUEST - invalid/incomplete request or body object
+        401:
+          description: UNAUTHORIZED - no/invalid Keycloak token
+        404:
+          description: Not found
+        500:
+          description: INTERNAL SERVER ERROR - server encountered unexpected condition
+    put:
+      tags:
+        - topic-controller
+      summary: 'Updates a topic [Authorization: Role: topic-admin]'
+      operationId: updateTopic
+      parameters:
+        - name: id
+          in: path
+          description: Topic ID
+          required: true
+          schema:
+            type: integer
+            format: int64
+      requestBody:
+        content:
+          'application/json':
+            schema:
+              $ref: '#/components/schemas/TopicDTO'
+      responses:
+        200:
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TopicDTO'
+        401:
+          description: UNAUTHORIZED - no/invalid Keycloak token
+        409:
+          description: CONFLICT - unique constraint validation fails
+        500:
+          description: INTERNAL SERVER ERROR - server encountered unexpected condition
+components:
+  schemas:
+    TopicDTO:
```

Comment:
I could not exactly remember what we said for hal structure adaptions for topics but i thought we will include hal here i just can´t remember when :-P